doc/adr/0009-container-image-tags.md

```diff
@@ -0,0 +1,52 @@
+# 9. Convention for tagging of container images
+
+Date: 2020-03-13
+
+## Status
+
+Proposed
+
+## Context
+
+Libero's products are assembled from product specific, general libero and external container images.
+
+Integration testing and reproducible deploys require uniquely identifiable image tags.
+
+As a starting point for users each product has an umbrella repo. This is not only used to collect product issues and documents. I also contains e.g. docker-compose files, helm charts that combine various components to a useable whole.
+
+
+## Decision
+
+### Tagging 
+
+- libero container images shall always be tagged with {branch-full_git_hash}
+- images shall receive semver tags when the git repo receives them
+- a semver tagging event bumps its parent tags
+
+Examples:
+
+- liberoadmin/reviewer-client:master-full_git_hash
+- liberoadmin/reviewer-client:dev-full_git_hash
+- liberoadmin/reviewer-client:1.2.43
+
+Example Workflow:
+
+1. head of `libero-reviewer` `master` moves  
+   the hash is `2d57c085e31cad0c81bd2a7db8dfe28d93aec99b`
+   1. new container image gets built
+   2. image gets pushed to registry as  
+      `liberoadmin/reviewer-client:master-2d57c085e31cad0c81bd2a7db8dfe28d93aec99b`  
+      its short digest is: `dd75d2e1bbfc`
```

Comment:
how is the digest calculated? could that maybe be the first _n_ characters of the git hash? (which would help to see that they are actually the same)

Reply:
The digest is calculated during container creation process i.e. it is independent from the git-hash.

Reply:
Okay, it sounded like this might be another tag. Is it related to image tagging strategy?

Reply:
(worth an explanation as we haven't used Docker image digests inside eLife at all)

Reply:
In a sense yes. I've tried to elaborate in my answer to your above questions.